### PR TITLE
Add score multiplier on mods footer button

### LINKED ISSUE #6951: Display score multiplier on mods button
**Describe the new feature:** Simply display the score multiplier as well as active mods on the mods button.

**Proposal designs of the feature:** 

image

### PR DESCRIPTION
**Pull request by GSculerlor**
Resolve #6951 

https://streamable.com/2keld

No Mod:
Screen Shot 2019-11-28 at 20 10 45
Increased Multiplier:
Screen Shot 2019-11-28 at 20 11 13
Decreased Multiplier:
Screen Shot 2019-11-28 at 20 11 00
Screen Shot 2019-11-28 at 20 11 32

### CODE REVIEW

File: osu.Game.Tests/Visual/UserInterface/TestCaseFooterButtonMods.cs [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            AddStep(@"Add Multiple Mods", () => changeMods(new Mod[] { new OsuModEasy(), new OsuModNoFail() }));
+            AddAssert(@"Check multiple mod multiplier", () => footerButtonMods.MultiplierText.Text.Equals(@"0.25x"));
+        }
+
+        [Test]
+        public void TestClearMultiplier()
+        {
+            AddStep(@"Add mods", () => changeMods(new Mod[] { new OsuModDoubleTime(), new OsuModFlashlight() }));
+            AddStep(@"Clear selected mod", () => changeMods(Array.Empty<Mod>()));
+            AddAssert(@"Check empty multiplier", () => footerButtonMods.MultiplierText.Text.Equals(string.Empty));
```
Comment (huoyaoyuan) [inline osu.Game.Tests/Visual/UserInterface/TestCaseFooterButtonMods.cs]: `IsNullOrEmpty`

File: osu.Game.Tests/Visual/UserInterface/TestCaseFooterButtonMods.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        public TestCaseFooterButtonMods()
+        {
+            Add(footerButtonMods = new TestFooterButtonMods());
+        }
+
+        [Test]
+        public void TestIncrementMultiplier()
+        {
+            AddStep(@"Add Hidden", () => changeMods(new Mod[] { new OsuModHidden() }));
+            AddAssert(@"Check Hidden multiplier", () => footerButtonMods.MultiplierText.Text.Equals(@"1.06x"));
```
Comment (huoyaoyuan) [inline osu.Game.Tests/Visual/UserInterface/TestCaseFooterButtonMods.cs]: Never call `Equals` on string. Use `==`.

**Review by huoyaoyuan — COMMENTED** [review]
`string.Equals` is a big difference between Java and C#. In C# `==` does the right thing.

File: osu.Game.Tests/Visual/UserInterface/TestSceneFooterButtonMods.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        public TestSceneFooterButtonMods()
+        {
+            Add(footerButtonMods = new TestFooterButtonMods());
+        }
+
+        [Test]
+        public void TestIncrementMultiplier()
+        {
+            AddStep(@"Add Hidden", () => changeMods(new Mod[] { new OsuModHidden() }));
+            AddAssert(@"Check Hidden multiplier", () => footerButtonMods.MultiplierText.Text == @"1.06x");
```
Comment (peppy) [inline osu.Game.Tests/Visual/UserInterface/TestSceneFooterButtonMods.cs]: comparing with strings is clumsy. the mod rates may change and we don't want to have to manually update all these tests.

**Review by peppy — DISMISSED** [review]

**Review by smoogipoo — DISMISSED** [review]
Seems good to me for the time being

**Review by smoogipoo — APPROVED** [review]

**Pull request merged**